### ISSUE TITLE
Update Cyppress Framework to latest version

### ISSUE THREAD

**Issue by withanage** [pkp/pkp-lib#8239]
**Describe the Issue**
- Functionalities introduced in later Cypress versions are necessary to facilitate certain functions, e.g. cross-domain testing
- As a maintenance task

**What application are you using?**
 OJS, OMP and  OPS 
**Additional information**
. - To introduce new cypress e2e configuration , a `cypress.config.js` has to be introduced.
-  [IntegrationFolder](https://docs.cypress.io/guides/references/migration-guide#integrationFolder) from cypress has to be changed to  [specPattern](https://docs.cypress.io/guides/references/configuration#e2e)
- to enable cross-domain [testing](https://github.com/pkp/orcidProfile/issues/216#), following settings has to be done into  cypress.config.js 
   -   `  experimentalSessionAndOrigin : true, "chromeWebSecurity": false
`

**Comment by withanage** [comment]
Hi @asmecher, here I have added the ticket for the cypress upgrade.

**Comment by asmecher** [comment]
@withanage, do you have this running locally (on your own machine)? If so, do you have a PR for the necessary changes?

**Comment by NateWr** [comment]
The content tests would benefit from being on Cypress 7.4.0+. In our current version, it's not possible to upload a file directly via the API, so when we create the content data, we have to use the UI to upload submission files. We'd save time  (and code) if we could use the API directly.

See: https://github.com/cypress-io/cypress/issues/1647

**Comment by withanage** [comment]
Hi @jonasraoni , @asmecher  and  @NateWr ,

Here is a quick  update on the current status of the Cypress update  to OJS  stable-3_3_0 branch.

#### What has been done so far

- npm update # update cypress explicitly
- move cypress.json to cypress.config.js and make the necessary changes
- further testing of *.spec.js to *.cv.js and other renaming to support cypress 10
- remove integrationFolder option from the [lib/pkp/tools/travis/run-tests.sh](https://github.com/pkp/pkp-lib/compare/5de2f5e6f7d6102f4ade90e8b3bb9db6af26a0df...c24419b9b31c4f87d81e7751c601ff7b16e51d99#diff-e1695817bb30277e52fc34a2e365b0014d79fe4e195cf8e13b11f64c4b5f2eed) 
- added options to support cross-domain testing
- So far I haven't had to make any changes to the commands in pkp/lib.

**PR :** https://github.com/pkp/ojs/pull/3588/files

#### Next steps

I will monitor the tests and update as needed in the coming days, e.g. cypress-file-upload etc. If there is anything you have done before that might be helpful to me, please lewt me know, otherwise I will continue with the tests and progessively fix anything I come across.

**Comment by withanage** [comment]
Update: tests are passing until  create sections.

**TODO:** ordering the tests propersly

**Comment by withanage** [comment]
Hi  @asmecher,

A  quick update on the cypress update

All the tests for the PR with the cypress 10 for the main branch have passed

https://github.com/pkp/ojs/pull/3588/

#### Summary

- npm update -D cypress@latest
- move cypress.json to cypress.config.js and make the necessary changes
- moving  *.spec.js to *.cv.js and other renamings  to support cypress 10
- remove integrationFolder option from the lib/pkp/tools/travis/run-tests.sh, here the seperate integration test section was deleted and the integration tests were moved to [specPattern](https://github.com/pkp/ojs/pull/3596/files#diff-5bae461d15457335f3d6285d35d3df2fd618e583eb7460c52cd8f0de69627abcR30), which allows multiple folders.
- added options to support cross-domain testing
- No changes to the cypress  js files  in pkp/lib.

**Comment by asmecher** [comment]
Thanks, @withanage, that's great! This will need to wait for several open branches/PRs to be merged first:
- @NateWr's work on the submission wizard (https://github.com/pkp/pkp-lib/issues/7191)
- Apparently @ewhanson has an open PR with some tests (do you have a link for that, Erik?)
- @defstat is working on some tests at https://github.com/pkp/pkp-lib/issues/7608

Once all 3 of these are merged, we can walk through the batch changes and get Cypress updated.

**Comment by ewhanson** [comment]
Hey @asmecher, I'm working on new tests in #8027.

**Comment by NateWr** [comment]
I think @ewhanson has several new tests in #8027. However, these are mostly new tests and not a major rewrite of existing tests.

**Comment by ewhanson** [comment]
> I think @ewhanson has several new tests in #8027. However, these are mostly new tests and not a major rewrite of existing tests.

That's right, but I'd say go ahead with the merge if there's nothing else blocking it. It shouldn't be too painful to get these updated, and #8027 continues to take more time than expected.

**Comment by asmecher** [comment]
Resolved via:
- https://github.com/pkp/pkp-lib/pull/8514
- https://github.com/pkp/ojs/pull/3680
- https://github.com/pkp/omp/pull/1290
- https://github.com/pkp/ops/pull/425
